### PR TITLE
fix: make lifi network fees accurates

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

...ish, with some ramifications.

As we are aware of, Li.Fi upstream gas estimates are off, and actual tend to be 2x+ the estimate. 
While leveraging upstream estimates makes sense as a convenience, our own estimates from node are actually correct, and we weren't using that.
This PR fixes that, although, obviously, this only works under one scenario: you have a wallet connected, and you're either trading a token you have already granted allowance for, or a native asset.

Tokens without allowance granted for will fall under the wrong upstream estimates, and work to fetch last/correct estimates from node and reflect that to the user at quote time will be done in https://github.com/shapeshift/web/issues/8270

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/8270

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low, estimates are currently entirely off.

## Testing

- With a wallet connected and a token you have an allowance for *or* a native token, network fees should closely match the final ones from wallet
- Without a wallet connected, or with a token you don't have an allowance for, network fees should be the same as in develop currently

### Engineering

- ^

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

- ^

## Screenshots (if applicable)

Tested against develop (right)

- Native token

Screenshot 2024-12-09 at 22 47 39

- Token with allowance granted

Screenshot 2024-12-09 at 22 51 06

- Token without allowance granted

Screenshot 2024-12-09 at 22 49 26

### CODE REVIEW

File: packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts
Comment (gomesalexandre) [inline packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts:1]: Review me with whitespaces off - added own estimates as main block, and current logic in catch clause

File: packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts [hunk 1]
```diff
@@ -28,59 +31,84 @@ export const getNetworkFeeCryptoBaseUnit = async ({
...
+    if (data === undefined) throw new Error('transactionRequest: data is required')
+    if (to === undefined) throw new Error('transactionRequest: to is required')
+    if (value === undefined) throw new Error('transactionRequest: value is required')
+
+    // Attempt own fees estimation if possible (approval granted, wallet connected with from addy to use for simulation)
+    const feeData = await evm.getFees({
+      adapter,
+      data,
+      to,
+      value: bn(value.toString()).toString(),
```
Comment (woodenfurniture) [inline packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts:50]: [suggestion]: Value is already a string

```suggestion
      value,
```

Comment (woodenfurniture) [reply packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts:50]: Unless the type system is lying (not unlikely)?

**Review by woodenfurniture — APPROVED** [review]
Works as expected, just one non-blocking suggestion if it seems useful.

File: packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts (hunk 1 above)
Comment (gomesalexandre) [reply packages/swapper/src/swappers/LifiSwapper/utils/getNetworkFeeCryptoBaseUnit/getNetworkFeeCryptoBaseUnit.ts]: Monkey saw monkey did 🐒 confirmed we're still gucci
c4a0ef820d

**Pull request merged**